### PR TITLE
Fixed / Added Option support for: String, i32, i64, f32, f64, uuid, date, and bool

### LINKED ISSUE #126: Option other basic GValue types?
In https://github.com/wolf4ood/gremlin-rs/blob/2e18b3cb09217a1c7f9645f98ce98a90a69acfab/gremlin-client/src/structure/value.rs#L410

it seems Option<String> is supported but omitted other basic types like Int32, Int64, etc.

Could those be added?

### PR DESCRIPTION
**Pull request by criminosis**
Closes #126

### CODE REVIEW

File: gremlin-client/src/structure/value.rs [hunk 1]
```diff
@@ -339,6 +339,7 @@ impl std::convert::TryFrom<GValue> for uuid::Uuid {
     fn try_from(value: GValue) -> GremlinResult<Self> {
         match value {
             GValue::Uuid(uid) => Ok(uid),
+            GValue::List(s) => from_list(s),
```
Comment (criminosis) [inline gremlin-client/src/structure/value.rs:342]: Having to add these GValue::List instances kind of surprised me. Seems like if we're being sent back a List element something should be wrong? But from stepping through with a debugger we end up here from the `impl_try_from_option` macro which means it's the GValue coming from Tinkerpop 🤔

File: gremlin-client/src/structure/value.rs [hunk 2]
```diff
@@ -400,11 +433,18 @@ macro_rules! impl_try_from_option {
...
-                Ok(res)
+                let res: $t = value.try_into()?;
+                Ok(Some(res))
             }
         }
     };
 }
 
 impl_try_from_option!(String);
+impl_try_from_option!(i32);
```
Comment (criminosis) [inline gremlin-client/src/structure/value.rs:444]: Ideally we'd be able to invoke this macro with some kind of loop over invoking the macro? I'm going to be repeating all these value types again for #127 I believe, so if there's a way to do a loop let me know.

File: gremlin-client/src/structure/value.rs (hunk 1 above)
Comment (criminosis) [reply gremlin-client/src/structure/value.rs:342]: Stepping through with the debugger a little earlier, basically before it creates the properties Map I then deserialize values out of it seems to flow from deserialize_map to deserialize_list to deserialize_XXX where XXX are the various deserialize methods in https://github.com/wolf4ood/gremlin-rs/blob/master/gremlin-client/src/io/serializer_v3.rs does that seem correct / working as intended @wolf4ood ?

Comment (wolf4ood) [reply gremlin-client/src/structure/value.rs:342]: Yeah it' seems strange that you had to add the list variant let me check :)

Comment (wolf4ood) [reply gremlin-client/src/structure/value.rs:342]: Probably it has something to do with multivalue properties but let me recall in my head. The try from stuff was added for doing automatic structs binding :)

Comment (criminosis) [reply gremlin-client/src/structure/value.rs:342]: It seemed like it was creating singleton lists in the debugger when I was walking through the steps and then unpacking the single item out of the list. I'm working on #127 now so I may have more insight as make my way through it.

Comment (wolf4ood) [reply gremlin-client/src/structure/value.rs:342]: If I remember correctly when doing value map step the map  contains the values in form of array for supporting multi-properties on vertex type

File: gremlin-client/src/structure/value.rs [hunk 3]
```diff
@@ -375,6 +378,36 @@ impl std::convert::TryFrom<GValue> for bool {
     }
 }
 
+impl std::convert::TryFrom<GValue> for f32 {
```
Comment (criminosis) [inline gremlin-client/src/structure/value.rs:381]: Just a heads up this is a duplicate implementation of the same code in https://github.com/wolf4ood/gremlin-rs/pull/131/files to keep the PRs independent for review and merging. One of the implementations will need to be removed.

File: gremlin-client/src/structure/value.rs [hunk 4]
```diff
@@ -375,6 +378,36 @@ impl std::convert::TryFrom<GValue> for bool {
...
+            GValue::List(s) => from_list(s),
+            _ => Err(GremlinError::Cast(format!(
+                "Cannot cast {:?} to f32",
+                value
+            ))),
+        }
+    }
+}
+
+impl std::convert::TryFrom<GValue> for f64 {
```
Comment (criminosis) [inline gremlin-client/src/structure/value.rs:396]: Just a heads up this is a duplicate implementation of the same code in https://github.com/wolf4ood/gremlin-rs/pull/131/files to keep the PRs independent for review and merging. One of the implementations will need to be removed.

**Comment by wolf4ood** [comment]
Thanks @criminosis  :)

**Pull request merged**